src/contacts/processors/impl/SearchRecordProcessor.java

```diff
@@ -0,0 +1,57 @@
+package contacts.processors.impl;
+
+import contacts.ConsoleReader;
+import contacts.model.Record;
+import contacts.processors.IActionProcessor;
+import contacts.processors.IProcessorFactory;
+import contacts.service.impl.IRecordService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SearchRecordProcessor implements IActionProcessor {
+
+    private final IRecordService recordService;
+    private final IProcessorFactory processorFactory;
+
+    public SearchRecordProcessor(IRecordService recordService, IProcessorFactory processorFactory) {
+        this.recordService = recordService;
+        this.processorFactory = processorFactory;
+    }
+
+    @Override
+    public boolean doAction() throws IOException, ClassNotFoundException {
+        String query = ConsoleReader.getStringFromConsole("Enter search query: ");
+        List<Record> resultsList = searchMatching(query);
+        System.out.println("Found " + resultsList.size() + " results:");
+        int index = 0;
+        for (Record record : resultsList) {
+            index++;
+            record.printListRecord(index);
+
+        }
+        int recordNumber = ConsoleReader.getIntFromConsole("Select a record: ");
```

Comment:
what if list is empty?

Reply:
User will see message: "Found 0 results:"